fiat-core/src/main/java/com/netflix/spinnaker/fiat/providers/BaseProvider.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.fiat.providers;
+
+import lombok.Data;
+import lombok.Setter;
+import org.springframework.beans.factory.annotation.Value;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class BaseProvider {
```

Comment:
Components like Gate and Orca could use this as well, maybe we can factor this out and include it there at some point as well?